Implement GH-41: Add minimum width gui

mewtaylor opened the pull request:
This implements #41. It adds a new component, `Dropdown`, which is a formatted version of the `Popover` element.

I also moved the `modes.js` file to a `lib` folder, and added `layout-constants` from gui to the lib folder as well. This move from `modes/modes.js` to `lib/modes.js` can be reverted if needed.

@carljbowman and I discussed using the `ContextMenu` for this, however that library is pre-built to pop up wherever there is a click, which is not the desired behavior for this feature.

We could feasibly move the styling for `.mod-menu-item` into its own component on dropdown if we want, but I wasn't sure if that was a good idea or not.

mewtaylor committed:
move modes to lib

mewtaylor committed:
fix validation

mewtaylor committed:
add min width layout for tablet

rschamp commented on package.json:
```diff
@@ -67,6 +67,7 @@
     "react-intl-redux": "0.6.0",
     "react-popover": "0.5.4",
     "react-redux": "5.0.5",
+    "react-responsive": "^3.0.0",
```
We should pin this since it's actively used by the application

mewtaylor committed:
pin react-responsive type

thanks @rschamp!

mewtaylor committed:
fix linting, tests

mewtaylor committed:
Add back in missing props

Whoops! Thanks @fsih

fsih commented on src/lib/layout-constants.js:
```diff
@@ -0,0 +1,3 @@
+export default {
+    fullSizeMinWidth: 1096
```
This one says 1096 and the other one says 1095. Is that right?

fsih replied:
Also noticed that things stick out a bit before min width UI kicks in, so we might want to do it sooner?
![image](https://user-images.githubusercontent.com/2855464/32572067-772ac6fc-c497-11e7-8352-5f29562058d1.png)

fsih commented on src/components/paint-editor/paint-editor.jsx:
```diff
@@ -342,9 +402,7 @@ PaintEditorComponent.propTypes = {
     name: PropTypes.string,
-    onCopyToClipboard: PropTypes.func.isRequired,
     onGroup: PropTypes.func.isRequired,
-    onPasteFromClipboard: PropTypes.func.isRequired,
```
I think you want to add these back too

fsih commented on src/components/paint-editor/paint-editor.jsx:
```diff
@@ -248,7 +308,7 @@ class PaintEditorComponent extends React.Component {
                             <InputGroup className={styles.modModeTools}>
                                 <ModeToolsComponent
-                                    onCopyToClipboard={this.props.onCopyToClipboard}
+                                    onCopyToClipboard={this.props.onCopyToClipboard}        
```
Nit: there are some extra spaces here

mewtaylor committed:
fix linting

thanks @fsih!

fsih commented:
If the 1096/1095 difference is intentional, then this PR looks good to me

mewtaylor committed:
Add font family to menu items

popovers need explicit font family set

mewtaylor committed:
squish buttons together a bit more on tablet

mewtaylor committed:
fine-tune the responsive design

squish some things together a bit more, including in desktop mode (in order to fit the coming “rotation point” button). Also turn canvas into 480x360, as it is in the mocks

paulkaplan commented:
@fsih I think this requires your re-review.

fsih reviewed (approved):
Sorry! Thanks for the reminder ping
